overlord/devicestate/devicemgr.go

```diff
@@ -711,6 +711,18 @@ func (m *DeviceManager) ensureSeeded() error {
 	var opts *populateStateFromSeedOptions
 	if m.preseed {
 		opts = &populateStateFromSeedOptions{Preseed: true}
+		if !release.OnClassic {
+			// XXX: needed by loadDeviceSeed(), otherwise Ensure fails with:
+			// "devicemgr: cannot seed: internal error: requested inconsistent device seed: 20220105 (was )"
+			unloadDeviceSeed(m.state)
+			systemLabel, err := systemForPreseeding()
+			if err != nil {
+				logger.Noticef("%v", err)
+				return err
+			}
+			opts.Mode = "install"
+			opts.Label = systemLabel
+		}
 	} else {
 		modeEnv, err := maybeReadModeenv()
```

Comment:
there is various code that depends on what is read from moodeenv (for example preloadGadget) so I'm a bit surprised there are no more changes needed about that, maybe we need to make maybeReadModeenv itself preseed aware?

Reply:
or mabye differently, does the preseed script write an initial modeenv?

Reply:
I've made modenv error our if read/written during preseeding.
Preseed script doesn't write to initial modeenv. Modeenv is set to "run" inside doRestartSystemToRunMode on first boot.